src/Documents.jl

```diff
@@ -134,6 +134,31 @@ immutable EvalNode
     result :: Any
 end
 
+# Navigation
+# ----------------------
+
+"""
+Element in the navigation tree of a document, containing navigation references
+to other page, reference to the [`Page`](@ref) object etc.
+"""
+type NavNode
+    page           :: Nullable{Compat.UTF8String} # unless null, should point to a valid Page in the document
```

Comment:
Which are the cases where this could be an invalid page as opposed to a "valid Page"?